### ISSUE TITLE
Jobs able to access other jobs variables.

### ISSUE THREAD

**Issue by blmhemu** [hashicorp/nomad#14996]
### Nomad version
1.4.1

### Operating system and Environment details
Ubuntu arm64
Did not touch any ACL stuff. (Neither enabled nor disabled explicitly in the config)

### Issue
Job `A` can access Job `B`s env variables.

### Reproduction steps
Run a job (say `echo`) with the following template 
```
    Templates:
    - DestPath: "${NOMAD_SECRETS_DIR}/env.vars"
      LeftDelim: "[["
      RightDelim: "]]"
      ChangeMode: "restart"
      Envvars: true
      EmbeddedTmpl: |
        [[- range nomadVarList ]]
          [[- with nomadVar .Path ]]
            [[- range .Tuples ]]
        [[ .K ]]="[[ .V ]]"
            [[- end ]]
          [[- end ]]
        [[- end ]]

```
Put some nomad variables at `nomad/jobs/echo`
Put some variables at `nomad/jobs/web`
Now if you printenv in the job, you can see both variables.

From the blog at https://www.hashicorp.com/blog/nomad-1-4-adds-nomad-variables-and-updates-service-discovery, I thought this behaviour is the default behaviour. Am I wrong or is this a bug ?

#### Expected Result
No access to other job variables.

#### Actual Result
Able to access other job variables.

**Comment by tgross** [comment]
Hi @blmhemu!

> Did not touch any ACL stuff. (Neither enabled nor disabled explicitly in the config)

You need to configure ACLs explicitly in Nomad in order for them to be enforced. See the [Security Model](https://developer.hashicorp.com/nomad/docs/concepts/security#secure-configuration) docs for more information on that. This applies to Variables as well. 

> From the blog at https://www.hashicorp.com/blog/nomad-1-4-adds-nomad-variables-and-updates-service-discovery, I thought this behaviour is the default behaviour. Am I wrong or is this a bug ?

The blog just assumes that you've got ACLs enabled, I think. We've got a tutorial being published in the next week or so that has lots of details that might be helpful, but yes you need to have ACLs enabled for Variables too.

(Note that there's a bug we're about to land where templates can List but not Read variable paths for other jobs in their own namespace, but this metadata isn't considered sensitive. That should go out early next week.)

**Comment by blmhemu** [comment]
Got it. Let me check if I can enable ACLs and retry this.

**Comment by tgross** [comment]
Right! Once ACLs are enabled jobs won't be able to read other jobs variable contents, and jobs will get automatic Read access to the variables they own.

**Comment by blmhemu** [comment]
Hey ! One more question, currently I am using the following template to source all env variables which the jobs has access to.
```
    [[- range nomadVarListSafe ]]
      [[- if nomadVarExists .Path ]]
        [[- with nomadVar .Path ]]
          [[- range .Tuples ]]
    [[ .K ]]="[[ .V ]]"
          [[- end ]]
        [[- end ]]
      [[- end ]]
    [[- end ]]
```

but it fails when rendering the template `Missing: nomad.var.list(@default.global)`. What would be the correct way to do this ?

**Comment by tgross** [comment]
Are there any variables available to this job in the default namespace? The `nomadVarListSafe` blocks if there's no variables available. From [the docs](https://developer.hashicorp.com/nomad/docs/job-specification/template#nomad-variables):

> The `nomadVarListSafe` function works identically to `nomadVarList`, but refuses to render the template if the variable list query returns blank/empty data.

**Comment by blmhemu** [comment]
Hey ! Yes there is a variable under `nomad/jobs/<job-name>` and I tried with both safe and non-safe versions. Still getting the same error. If I enter the prefix `[[- range nomadVarListSafe "nomad/jobs" ]]`, it works.

**Comment by blmhemu** [comment]
Cool ! Thanks for the swift follow up. I have a small devx suggestion though. Instead of adding a template like below, I suggest we add a stanza in hcl / json like `source_nomad_vars: true` which does it automatically.
```
[[- range nomadVarListSafe ]]
  [[- if nomadVarExists .Path ]]
    [[- with nomadVar .Path ]]
      [[- range .Tuples ]]
[[ .K ]]="[[ .V ]]"
      [[- end ]]
    [[- end ]]
  [[- end ]]
[[- end ]]
```

**Comment by tgross** [comment]
Fixes have shipped in Nomad 1.4.2 

>  I have a small devx suggestion though. Instead of adding a template like below, I suggest we add a stanza in hcl / json like source_nomad_vars: true which injects the relevant env variables.

@blmhemu that would be a nice feature. I've opened https://github.com/hashicorp/nomad/issues/15064 for further discussion of that.

**Comment by blmhemu** [comment]
Awesome ! Verified on nomad 1.4.2. Thanks @tgross .
